src/webgl/interaction.js

```diff
@@ -93,14 +93,18 @@ p5.prototype.orbitControl = function(sensitivityX, sensitivityY, sensitivityZ) {
 
   const zoomScaleFactor = 0.02;
   const scaleFactor = this.height < this.width ? this.height : this.width;
+  const uP = this._renderer.uPMatrix.mat4;
```

Comment:
Do you think maybe we should edit `this._renderer.uPMatrix.mat4` directly instead of making a new variable referencing the same object? The result is the same either way, but it might be clearer to readers that we're modifying the matrix directly.

Reply:
If it's only used here, that's fine, but I think it might be better to prepare it as a variable in advance because it will be used later when creating a method for moving with the right mouse button.

Reply:
That idea is to drop the gaze point into the normalized device coordinate system using the camera matrix and the projection matrix, move the mouse by the inverse of the movement distance, and focus on the position where the normalized device coordinates come to the destination.
It shifts the viewpoint by parallel movement.
This allows the movement of the camera along with the movement of the mouse. I don't use the modelview matrix because it doesn't involve model transformations. I've already confirmed that it works fine with perspective, ortho, and frustum.
demo:[improve panning](https://editor.p5js.org/dark_fox/sketches/fxWMOHeEE)

Reply:
Ahh I see, I thought you meant you would be updating the projection matrix to implement movement, I see what you mean now -- this is what I was suggesting too.

Reply:
I think it might still be more readable if we declare `uP` just in that branch for that case, as we're just reading it. Then, here, where we're actually modifying it, it's clear to readers that the `uPMatrix` property will be modified.

Reply:
I got it. So, let's write this directly and declare it again in the method I will implement later.